elasticdl/python/worker/worker.py

```diff
@@ -220,10 +222,17 @@ def get_model_from_master(self, version, method):
 
     def get_model_from_ps(self, version, method):
         model_version = -1
-        for stub in self._ps_stubs:
+        for ps_id, stub in enumerate(self._ps_stubs):
+            if ps_id not in self._ps_vars:
+                continue
             req = empty_pb2.Empty()
             res = stub.pull_variable(req)
 
+            if not res.model_init_status:
+                # push variable to ps for initialization
+                self.report_variable_to_ps(ps_id)
+                res = stub.pull_variable(req)
```

Comment:
Do we need to check `res.model_init_status` here?

Reply:
Add raise exception and comment to indicate that we should handle these exceptions when dealing with ps fault-tolerance.